library/src/scala/runtime/EnumValueSerializationProxy.java

```diff
@@ -0,0 +1,40 @@
+package scala.runtime;
+
+import java.io.Serializable;
+import java.security.AccessController;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+
+/** A serialization proxy for singleton enum values */
+public final class EnumValueSerializationProxy implements Serializable {
+    private static final long serialVersionUID = 1L;
+    private final Class<?> enumClass;
+    private final int ordinal;
+    private static final ClassValue<Object[]> enumValues = new ClassValue<Object[]>() {
+        @Override
+        protected Object[] computeValue(Class<?> type) {
+            try {
+                return AccessController.doPrivileged((PrivilegedExceptionAction<Object[]>) () ->
+                  (Object[])type.getMethod("values").invoke(null));
+            } catch (PrivilegedActionException e) {
+                return rethrowRuntime(e.getCause());
+            }
+        }
+    };
+
+    private static <T> T rethrowRuntime(Throwable e) {
```

Comment:
This looks almost like the rethrowRuntime method in ModuleSerializationProxy, but this one is generic, is there any reason for it to be different? https://github.com/scala/scala/blob/2.13.x/src/library/scala/runtime/ModuleSerializationProxy.java
(Also would be good to add a comment at the top of this file mentioning that it's based on ModuleSerializationProxy)

Reply:
im not really sure what's the use of it being a separate method, I'll inline it and add the comment